RecoTracker/SpecialSeedGenerators/python/SimpleCosmicBONSeeder_cfi.py

```diff
@@ -29,8 +27,34 @@ def makeSimpleCosmicSeedLayers(*layers):
     #print "SEEDING LAYER LIST = ", layerList
     return layerList
 
-layerInfo = RecoTracker.SpecialSeedGenerators.CombinatorialSeedGeneratorForCosmics_cfi.layerInfo.clone(
-    TEC = dict(useSimpleRphiHitsCleaner = False)
+layerInfo = dict(
```

Comment:
Why not keeping cloning from RecoTracker.SpecialSeedGenerators.CombinatorialSeedGeneratorForCosmics_cfi, instead of all this copy/paste?

Reply:
@perrotta : I got an error when I keep L32-33 as below, so I did simply copy/paste the layerInfo in this SimpleCosmicVONSeeder_cfi.py file. 
Please let me know if there's the most suitable way you suggest here. 
```
  File "/afs/cern.ch/work/j/jelee/ServiceWork/NewTask/0607/CMSSW_12_0_X_2021-06-06-2300/python/RecoTracker/SpecialSeedGenerators/SimpleCosmicBONSeeder_cfi.py", line 30, in <module>
    layerInfo = RecoTracker.SpecialSeedGenerators.CombinatorialSeedGeneratorForCosmics_cfi.layerInfo.clone(
AttributeError: 'dict' object has no attribute 'clone'
```

Reply:
Then probably the simplest solution is to revert the change in https://github.com/cms-sw/cmssw/pull/34009/files#diff-0dbbc624124b7539ebca54a0a8940ad7ca30d06b687c5bbb825c30a51e16dd7eR9

Reply:
@makortel , perhaps you can help here...

I would like to avoid copy/pasting all the PSet's below, which were already defined in RecoTracker/SpecialSeedGenerators/python/CombinatorialSeedGeneratorForCosmics_cfi.py.  This would be inconvenient in case of modifications in the source config, that couldn't be automatically propagated to the copy/pasted ones.

After investigating a bit how do dictionaries and unpacking operatoris work in python, I came to the conclusion that the simplest way to avoid it is to simply revert the changes for `layerInfo` here and in  https://github.com/cms-sw/cmssw/pull/34009/files#diff-0dbbc624124b7539ebca54a0a8940ad7ca30d06b687c5bbb825c30a51e16dd7eR9

But perhaps python allows some more elegant and correct solution, that allows keep defining `layerInfo` as a dict in the source file, and propagate to the derived config with the possibility to modify a parameter (which is what causes the issue here). Do you have any suggestion?

Reply:
With `dict` you can use `copy` module, i.e. along
```py
import copy
dict_copy = copy.deepcopy(dict_orig)
```

For this specific case I'd like to remind that we recently allowed passing `PSet` as the first argument(s) for `clone()`. I think you could keep the `layerInfo` definition here as it is, and change `SimpleCosmicBONSeeder_cff.py` to something along
```py
simpleCosmicBONSeedingLayers = _mod.seedingLayersEDProducer.clone(
    layerInfo,
    layerList = layerList
)
```

Reply:
Thank you @makortel 
And what about modifying a parameter in an included PSet, as it was done originally in https://github.com/cms-sw/cmssw/pull/34009/files#diff-f633f65b880b125f9f08753ab0640fef555f76d83ce539433c8bcfeb3baecdc5L33 ?

Reply:
I'm not sure I follow your question. The original code has
```py
layerInfo = RecoTracker.SpecialSeedGenerators.CombinatorialSeedGeneratorForCosmics_cfi.layerInfo.clone(
    TEC = dict(useSimpleRphiHitsCleaner = False)
)
```
that could stay as it is (the modification is in a `clone()` call after all).

Reply:
Ok, Matti, thank you!

Then, if I understand it correctly, the suggestion for @jeongeun is:
- Revert the change in  RecoTracker/SpecialSeedGenerators/python/CombinatorialSeedGeneratorForCosmics_cfi.py and let the `layerInfo` as a PSet
- Modify `simpleCosmicBONSeedingLayers` as you write [above](https://github.com/cms-sw/cmssw/pull/34009/files#r649176072)
- Revert also the definition  of `layerInfo` here in SimpleCosmicBONSeeder_cfi.py, with a clone and only the modified parameter replaced in it

Reply:
Thank you all!
To be consistent, I will revert (`dict` -> `cms.PSet`)all the changes in these 3 cfi.py files RecoTracker/SpecialSeedGenerators/python/CombinatorialSeedGeneratorForBeamHalo_cfi.py, 
RecoTracker/SpecialSeedGenerators/python/CombinatorialSeedGeneratorForCosmics_cfi.py, RecoTracker/SpecialSeedGenerators/python/CombinatorialSeedGeneratorForCosmicsRegionalReconstruction_cfi.py
Then, I will revert `**layerInfo`  ->  `layerInfo` as it is.